Src/AutoNSubstitute/AutoNSubstitute.csproj

```diff
@@ -64,6 +64,7 @@
   <ItemGroup>
     <Compile Include="AutoConfiguredNSubstituteCustomization.cs" />
     <Compile Include="AutoNSubstituteCustomization.cs" />
+    <Compile Include="AutoSubstituteCustomization.cs" />

```

Comment:
> rename the new `SubstituteAttribute` to `NSubstituteAttribute`

I don't particularly mind, but this type is, IIRC, the original use case that this pull request is all about. Originally, you wanted to be able to write `[Substitute]foo`. With the proposed change, you'd have to write `[NSubstitute]foo`. Do you think that's better?

> rename the new `SubstituteAttributeRelay` to `NSubsituteAttributeRelay`

If you rename `SubstituteAttribute` to `NSubstituteAttribute`, then this makes sense as well. If you decide not to rename `SubstituteAttribute` to `NSubstituteAttribute`, then I also think the name should remain `SubstituteAttributeRelay`, because it relays that particular attribute.

> rename the new `SubstituteRequest` to `NSubstituteRequest`

I don't know... Are other objects requesting a *substitute*, or an *nsubstitute*?

> `[Obsolete]` the existing `NSubstituteBuilder` which is being replaced by the new `NSubstituteAbstractTypeRelay`

Is it necessary to do this as part of this pull request?